src/main/java/org/javarosa/form/api/FormEntryPrompt.java

```diff
@@ -95,63 +94,52 @@ public IAnswerData getAnswerValue() {
                         String value = itemset.getRelativeValue().evalReadable(form.getMainInstance(), new EvaluationContext(form.getEvaluationContext(), node.getRef()));
                         preselectedValues.add(value);
                     }
-                } else {
-               List<Selection> sels;
-                    IAnswerData data = mTreeElement.getValue();
-                    if (data instanceof MultipleItemsData) {
-                        sels = (List<Selection>)data.getValue();
-                    } else if (data instanceof SelectOneData) {
-                        sels = new ArrayList<Selection>(1);
-                        sels.add((Selection)data.getValue());
-                    } else {
-                        sels = new ArrayList<Selection>(0);
-                    }
-                    for (int i = 0; i < sels.size(); i++) {
-                        preselectedValues.add(sels.get(i).xmlValue);
-                    }
-                }
 
-                //populate 'selection' with the corresponding choices (matching 'value') from the dynamic choiceset
-            List<Selection> selection = new ArrayList<Selection>();
-                for (int i = 0; i < preselectedValues.size(); i++) {
-                    String value = preselectedValues.get(i);
-                    SelectChoice choice = null;
-                    for (int j = 0; j < choices.size(); j++) {
-                        SelectChoice ch = choices.get(j);
-                        if (value.equals(ch.getValue())) {
-                            choice = ch;
-                            break;
+                    //populate 'selection' with the corresponding choices (matching 'value') from the dynamic choiceset
+                    List<Selection> selection = new ArrayList<Selection>();
+                    for (int i = 0; i < preselectedValues.size(); i++) {
+                        String value = preselectedValues.get(i);
+                        SelectChoice choice = null;
+                        for (int j = 0; j < choices.size(); j++) {
+                            SelectChoice ch = choices.get(j);
+                            if (value.equals(ch.getValue())) {
+                                choice = ch;
+                                break;
+                            }
+                        }
+                        //if it's a dynamic question, then there's a good choice what they selected last time
+                        //will no longer be an option this go around
+                        if (choice != null) {
+                            selection.add(choice.selection());
                         }
                     }
-                    //if it's a dynamic question, then there's a good choice what they selected last time
-                    //will no longer be an option this go around
-                    if(choice != null)
-                    {
-                        selection.add(choice.selection());
```

Comment:
I think this used to ensure there was always a `SelectionChoice` for the `Selection`.